parsec/core/mountpoint/manager.py

```diff
@@ -134,6 +136,55 @@ def get_path_in_mountpoint(
         await self._mountpoint_tasks[(workspace_id, timestamp)].cancel_and_join()
         del self._mountpoint_tasks[(workspace_id, timestamp)]
 
+    async def remount_workspace_new_timestamp(
+        self, workspace_id: EntryID, original_timestamp: Pendulum, target_timestamp: Pendulum
+    ) -> PurePath:
+        """
+        Mount the workspace at target_timestamp, and unmount the workspace at the original
+        timestamp if it is not None. If both timestamps are equals, do nothing.
+        """
+        # TODO : use different workspaces for temp mount
+        if original_timestamp == target_timestamp:
+            try:
+                return self._mountpoint_tasks[(workspace_id, target_timestamp)].value
+            except KeyError:
+                pass
+            try:
+                self.user_fs.get_workspace(workspace_id)
+                raise MountpointNotMounted(
+                    f"Workspace `{workspace_id}` not mounted at timestamped `{target_timestamp}`"
+                )
+            except FSWorkspaceNotFoundError as exc:
+                raise MountpointConfigurationError(
+                    f"Workspace `{workspace_id}` doesn't exist"
+                ) from exc
+        if original_timestamp is None:
+            workspace = self._get_workspace(workspace_id)
+        else:
+            try:
+                workspace = self._get_workspace_timestamped(workspace_id, original_timestamp)
+            except MountpointNotMounted:
+                workspace = self._get_workspace(workspace_id)
+        try:
+            new_workspace = await workspace.to_timestamped(target_timestamp)
+        except FSWorkspaceTimestampedTooEarly as exc:
+            raise MountpointConfigurationWorkspaceFSTimestampedError(
+                f"Workspace `{workspace_id}` didn't exist at `{target_timestamp}`",
+                workspace_id,
+                target_timestamp,
+                workspace.get_workspace_name(),
+            ) from exc
+        self._timestamped_workspacefs[(workspace_id, target_timestamp)] = new_workspace
```

Comment:
This could also be a helper: 
```python
    async def _load_workspace_timestamped(self, workspace_id: EntryID, timestamp: Pendulum, original_timestamp: Optional[Pendulum] = None):
        workspace = self._get_workspace(workspace_id)
        try:
            new_workspace = await workspace.to_timestamped(target_timestamp)
        except FSWorkspaceTimestampedTooEarly as exc:
            raise MountpointConfigurationWorkspaceFSTimestampedError(
                f"Workspace `{workspace_id}` didn't exist at `{target_timestamp}`",
                workspace_id,
                target_timestamp,
                workspace.get_workspace_name(),
            ) from exc
        self._timestamped_workspacefs[(workspace_id, target_timestamp)] = new_workspace
        return new_workspace